lib/matplotlib/colors.py

```diff
@@ -532,6 +532,7 @@ def __call__(self, X, alpha=None, bytes=False):
         # otherwise the under-range values get converted to over-range.
         xa[xa > self.N - 1] = self._i_over
         xa[xa < 0] = self._i_under
+        xa[np.isnan(X)] = self._i_bad
```

Comment:
This triggers a test failure: `FutureWarning: in the future, boolean array-likes will be handled as a boolean array index`.  I guess X is a list for these and needs to be an array?

Reply:
Interesting... That is actually the behavior I want for this case. This is warning because it is using numpy version 1.11, newer versions are just fine as far as I can tell. The failure is actually because of scalars being passed in rather than lists I think. It is producing a FutureWarning because there are 'good' values currently, but this can produce an error in the situation where a single nan is passed in `cmap(np.nan)`:
```
X = np.nan
xa = np.array([X])
xa[np.isnan(X)] # errors because it tries to access the first index, which doesn't exist
```
Right now, this works as expected for newer versions of numpy. Let me think if there is a better way of doing this...